apps/rush-lib/src/logic/taskExecution/ProjectLogWritable.ts

```diff
@@ -28,24 +27,28 @@ export class ProjectLogWritable extends TerminalWritable {
       this._project.packageName
     );
 
-    this._buildLogPath = path.join(this._project.projectFolder, `${unscopedProjectName}.build.log`);
-    this._errorLogPath = path.join(this._project.projectFolder, `${unscopedProjectName}.build.error.log`);
+    this._logPath = `${this._project.projectFolder}/${unscopedProjectName}.${logFilename}.log`;
+    this._errorLogPath = `${this._project.projectFolder}/${unscopedProjectName}.${logFilename}.error.log`;
 
-    FileSystem.deleteFile(this._buildLogPath);
+    FileSystem.deleteFile(this._logPath);
     FileSystem.deleteFile(this._errorLogPath);
 
-    this._buildLogWriter = FileWriter.open(this._buildLogPath);
+    this._logWriter = FileWriter.open(this._logPath);
+  }
+
+  public static normalizeCommandNameForLogFilenames(commandName: string): string {
+    return commandName.replace(/[^a-zA-Z0-9]/g, '_');
```

Comment:
This seems overengineered. Why can't we simply enforce that the `commandName` is a valid filename part?  This would eliminate the need for normalization and also eliminate the possiblity of collisions.

Reply:
Updated to just replace `:` with `_`